cpp/unreal_plugins/CoreUtils/Source/CoreUtils/Unreal.h

```diff
@@ -365,4 +377,29 @@ class COREUTILS_API Unreal
             return vec.at(0);
         }
     }
+
+    //
+    // Helper function to get fully qualified component names.
+    //
+
+    static std::string getFullyQualifiedComponentName(USceneComponent* component, const std::string& separator, bool include_actor_name = false)
+    {
+        SP_ASSERT(component);
+        SP_ASSERT(component->GetOwner());
+
+        std::string name = toStdString(component->GetName());
+
+        TArray<USceneComponent*> parents;
+        component->GetParentComponents(parents);
+        for (auto& parent : parents) {
```

Comment:
`auto` instead of `auto&` because we use `auto` for pointer types

Reply:
Done